04_numeronym/pfannerj/main.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var mainout io.Writer = os.Stdout
+
+func main() {
+
+	fmt.Fprintln(mainout, numeronyms("accessibility", "Kubernetes", "abc"))
+
+}
+
+func numeronyms(vals ...string) []string {
+
+	valsOut := vals
+	for i := range vals {
+		//Regular expression can be changed or removed altogether if additional chars need to be included
+		regEx := regexp.MustCompile("[^a-zA-Z0-9]+")
+		alphaNumVal := regEx.ReplaceAllString(vals[i], "") //Strips all non-alpha-numeric characters
+		if len(alphaNumVal) > 3 {
+			alphaNumChar := strings.Split(alphaNumVal, "")
+			valsOut[i] = alphaNumChar[0] + strconv.Itoa(len(alphaNumVal)-2) + alphaNumChar[len(alphaNumVal)-1]
```

Comment:
Here you can simplify this, there shouldn't be a need to use the `strings.Split()` method, you should be able to get the values directly from the `alphaNumVal`. If you use the `fmt.Sprintf()` method as well you can do something like below:

```go
length := len(alphaNumVal)
if length > 3 {
    valsOut[i] = fmt.Sprintf("%v%d%v", string(alphaNumVal[0]), length-2, string(alphaNumVal[length-1]))
}
```

And if you combine this with also removing the regex statements above you can simplify the logic heaps by using the value returned from the `range` function.

```go
func numeronyms(vals ...string) []string {
    output := vals
    for i, v := range vals {
        length := len(v)
        if length > 3 {
            output[i] = fmt.Sprintf("%v%d%v", string(v[0]), length-2, string(v[length-1]))
        }
    }
    return output
}
```

Reply:
Oh, that's heaps better... I thought I needed to do the string.Split in order to be able to refer to the [0] values etc.  Will definitely make these changes.  Thx!